cmd/opm/alpha/template/composite.go

```diff
@@ -33,166 +26,52 @@ and a 'composite template' file`,
 and a 'composite template' file`,
 		Args: cobra.MaximumNArgs(0),
 		Run: func(cmd *cobra.Command, args []string) {
-			containerTool = "docker"
-			var tempCatalog io.ReadCloser
-			catalogURI, err := url.ParseRequestURI(catalogFile)
-			if err != nil {
-				tempCatalog, err = os.Open(catalogFile)
-				if err != nil {
-					log.Fatalf("opening catalog config file %q: %v", catalogFile, err)
-				}
-				defer tempCatalog.Close()
-			} else {
-				tempResp, err := http.Get(catalogURI.String())
-				if err != nil {
-					log.Fatalf("fetching remote catalog config file %q: %v", catalogFile, err)
-				}
-				tempCatalog = tempResp.Body
-				defer tempCatalog.Close()
-			}
-			catalogData := tempCatalog
-
-			// get catalog configurations
-			catalogConfig := &composite.CatalogConfig{}
-			catalogDoc := json.RawMessage{}
-			catalogDecoder := yaml.NewYAMLOrJSONDecoder(catalogData, 4096)
-			err = catalogDecoder.Decode(&catalogDoc)
-			if err != nil {
-				log.Fatalf("decoding catalog config: %v", err)
-			}
-			err = json.Unmarshal(catalogDoc, catalogConfig)
-			if err != nil {
-				log.Fatalf("unmarshalling catalog config: %v", err)
-			}
 
-			if catalogConfig.Schema != composite.CatalogSchema {
-				log.Fatalf("catalog configuration file has unknown schema, should be %q", composite.CatalogSchema)
+			switch output {
+			case "yaml":
+				// do nothing
+			case "json":
+				// do nothing
+			default:
+				log.Fatalf("invalid --output value %q, expected (json|yaml)", output)
```

Comment:
+1

Reply:
I'll push back on this one, because I noticed it too but made the conscious choice to perpetuate a common pattern which was also demonstrated in
cmd/opm/init/cmd.go
cmd/opm/render/cmd.go
cmd/opm/alpha/template/basic.go

Reply:
Right, but in all three of those cases there is custom logic for the yaml and json cases vs a no-op.
e.g. https://github.com/operator-framework/operator-registry/blob/master/cmd/opm/alpha/template/basic.go#L41-L48

Reply:
+1 to @oceanc80's point

Reply:
Not a huge deal though, so if you prefer leaving this as is I'm okay with it